packages/core/src/lib/markdown-it/highlight/HighlightRuleComponent.ts

```diff
@@ -8,28 +8,33 @@ const UNBOUNDED = -1;
 export class HighlightRuleComponent {
   lineNumber: number;
   isSlice: boolean;
-  bounds: number[][];
-  constructor(lineNumber: number, isSlice: boolean = false, bounds: number[][] = []) {
+  bounds: Array<[number, number]>;
+
+  constructor(lineNumber: number, isSlice: boolean = false, bounds: Array<[number, number]> = []) {
     this.lineNumber = lineNumber;
     this.isSlice = isSlice;
     this.bounds = bounds;
   }
 
+  static isValidLineNumber(lineNumStr: string, min: number, max: number, offset: number) {
+    let lineNum = Number(lineNumStr);
+    if (Number.isNaN(lineNum) || !Number.isInteger(lineNum)) return null;
+    lineNum += offset;
+    return lineNum >= min && lineNum <= max ? lineNum : null;
```

Comment:
Will this handle the case where offset is negative? e.g. offset = -1, lineNum = 1, lineNum -= 1 and therefore lineNum < min?

I suppose max is not affected, but min might be problematic?

Reply:
In the case you propose wouldn't it be the function caller's responsibility to pass in the the right min, and yes it already caters to negative offset, since most offsets are negative, when the author puts a greater than 1 line start number of the code block?

Reply:
> In the case you propose wouldn't it be the function caller's responsibility to pass in the the right min

Yes I agree? On line 33 and line 50, I see that we are calling this function with "1". My concern is will "1" work for all the cases?

Reply:
Correct me if I interpret wrongly, hmm but shouldn't and wouldn't max be 3 in this case? Of which 4 exceeds hence no highlight applied being invalid since line doesn't exist which is the ideal behaviour? The screenshot seems consistent with the expected behaviour?

Reply:
I'm not saying this is the validate line number method's problem, but do we need to deal with the `''` in the array when invoking this method at the caller side

Reply:
Hmm, I see so for all code blocks it auto appends a '' to the end? is that what you mean

Reply:
The output is ok.

> u want to find out why an extra "" is appended to a code block am I correct to say that?

Not exactly. I believe the extra "" is appended because we split by "\n", something like:
![image](https://github.com/MarkBind/markbind/assets/41845017/5ab364ad-73a5-4d5c-bb0f-c0c9182b7594)

Would like to get your help to check on how we invoke the highlight method which in turn invoke the validate method in index.ts:
![image](https://github.com/MarkBind/markbind/assets/41845017/0f668abf-2a7e-46b2-97ae-d321b8e8bca7)

Question:
- should we also do a lines.pop() before we start highlighting? This is so that when we call the validate line number, we are actually passing the right "max".

@zlimez sorry for this additional check, this should be the last thing for this PR (hopefully:))

Reply:
Make sense will include

Reply:
Done and updated the test cases